test/functional/provider/admin/messages/inbox_controller_test.rb

```diff
@@ -48,4 +48,24 @@ def test_index_system_message
     assert_select 'title', "Inbox - Index | Red Hat 3scale API Management"
     assert_select '#export-to-csv', false, 'Export all Messages'
   end
+
+  test "creates valid reply" do
+    login_as(@admin)
+    post :reply, params: { message: { subject: "Valid Message", :body => "message with subject" }, id: @message.id }
+
+    MessageWorker.drain
+    msg = Message.last
```

Comment:
This could cause an intermittent failed test, in case we're running tests in parallel and another unexpected message becomes the last one.
We could change this to
```ruby
message = Message.find_by(sender: @message.receiver)
```
instead to make it safer.

Reply:
> instead we can do this
> msg = Message.find_by(id: @message.message_id)

You can't because that would return the original message and we want the reply.

Reply:
yes I think getting last message is the correct approach we have used the same in other test cases as well

Reply:
Fine by me :v: